eclipse-scout-core/src/desktop/outline/pages/js/JsPageHelper.ts

```diff
@@ -0,0 +1,470 @@
+    this.page.childrenLoaded = false;
+
+    // clean up _childPagesById map
+    this._childPagesById.clear();
+
+    if (!this.page.detailTable) {
+      return;
+    }
+
+    // remove all rows, otherwise e.g. double-clicking a row will lead to errors as the page was deleted
+    this.page.detailTable.deleteAllRows();
+  }
+
+  protected _installPageWithTableListeners() {
+    if (!(this.page instanceof PageWithTable)) {
+      return;
+    }
+
+    this.page.outline.on('nodesInserted', this._nodesInsertedHandler);
+
+    // detailTable may change -> install for current and listen on propertyChange
+    this._installPageWithTableTableListeners(this.page?.detailTable);
+    this.page.on('propertyChange:detailTable', e => {
+      this._uninstallPageWithTableTableListeners(e.oldValue);
+      this._installPageWithTableTableListeners(e.newValue);
+    });
+  }
+
+  protected _uninstallPageWithTableListeners() {
+    if (!(this.page instanceof PageWithTable)) {
+      return;
+    }
+
+    this.page.outline.off('nodesInserted', this._nodesInsertedHandler);
+    this._uninstallPageWithTableTableListeners(this.page?.detailTable);
+  }
+
+  protected _onNodesInserted(event: TreeNodesInsertedEvent) {
+    if (event.parentNode !== this.page) {
+      return;
+    }
+
+    // The event is triggered twice during the load/reload of a PageWithTable that creates its child pages using the JsPageHelper.
+    // 1. When the child pages are inserted on the UI server it is called for the first time.
+    //    In this case e.g. the text may not be correct as the page was not updated from the table row yet.
+    //    If this happens The PageWithTable is marked with childrenLoaded=false.
+    // 2. The second time this event is triggered is when the PageWithTable replaces its rows.
+    //    Now the child page is ready to be inserted as it was linked to the row and updated from it.
+    //    As the table load/reload is now completed the PageWithTable is marked with childrenLoaded=true.
+
+    // Mark all child pages loading iff the PageWithTable has not loaded its children.
+    for (const childPage of event.nodes) {
+      childPage.setCssClass(styles.toggleCssClass(childPage.cssClass, 'js-page-child-page-loading', !this.page.childrenLoaded));
+      childPage._decorate();
+    }
+  }
+
+  protected _installPageWithTableTableListeners(table: Table) {
+    if (!table) {
+      return;
+    }
+    table.on('rowsInserted', this._tableRowsInsertHandler);
+  }
+
+  protected _uninstallPageWithTableTableListeners(table: Table) {
+    if (!table) {
+      return;
+    }
+    table.off('rowsInserted', this._tableRowsInsertHandler);
+  }
+
+  protected _onTableRowsInserted(event: TableRowsInsertedEvent) {
+    // When rows are inserted in a PageWithTable, all child pages are ready, i.e. they are linked to their row and updated from it.
+    // Send a nodesChanged event in order to send summary information from parent PageWithTable to the UI server.
```

Comment:
```suggestion
    // Send a nodesChanged event in order to send summary information from the PageWithTable to the UI server.
```
I think the word "parent" might be confusing here (there is only one PageWithTable involved).